packages/iTwinUI-react/src/core/utils/hooks/useDragAndDrop.tsx

```diff
@@ -11,24 +13,64 @@ import React from 'react';
  * `onPointerDown` - handler that is called when pointer is down and handles all the dragging logic.
  * `transform` - current transform of the element, it is used to preserve drag position when element visibility is being toggled.
  */
-export const useDragAndDrop = (elementRef: React.RefObject<HTMLElement>) => {
+export const useDragAndDrop = (
+  elementRef: React.RefObject<HTMLElement>,
+  containerRect?: DOMRect,
+) => {
   const grabOffsetX = React.useRef(0);
   const grabOffsetY = React.useRef(0);
   const translateX = React.useRef(0);
   const translateY = React.useRef(0);
 
+  const distancesToEdges = React.useRef({
+    top: 0,
+    right: 0,
+    bottom: 0,
+    left: 0,
+  });
+
+  const containerRectRef = useLatestRef({
+    top: containerRect?.top ?? 0,
+    right: containerRect?.right ?? getWindow()?.innerWidth ?? 0,
+    bottom: containerRect?.bottom ?? getWindow()?.innerHeight ?? 0,
+    left: containerRect?.left ?? 0,
+  });
+
   const [transform, setTransform] = React.useState('');
 
-  const onPointerMove = React.useCallback(
-    (event: PointerEvent) => {
-      if (elementRef.current) {
-        translateX.current = event.clientX - grabOffsetX.current;
-        translateY.current = event.clientY - grabOffsetY.current;
-        elementRef.current.style.transform = `translate(${translateX.current}px, ${translateY.current}px)`;
-      }
-    },
-    [elementRef],
-  );
+  const onPointerMove = React.useRef((event: PointerEvent) => {
```

Comment:
why using `ref`? Why didnt make same with the callback below ?

Reply:
Just to avoid bugs and memory leaks when deps changes and this function is not being removed from event listeners.